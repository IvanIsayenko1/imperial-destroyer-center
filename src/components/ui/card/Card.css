.card {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  height: 100%;
  background: rgba(0, 0, 0, 0.3);
  backdrop-filter: blur(10px);
  border: 1px solid rgba(255, 59, 59, 0.2);
  transition: transform 0.3s ease;
}

.card:hover {
  transform: translateY(-5px);
  box-shadow: 0 0 20px rgba(255, 59, 59, 0.2);
}

.image {
  width: 100%;
  aspect-ratio: 16/9;
  overflow: hidden;
}

.image img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  transition: transform 0.3s ease;
}

.card:hover .image img {
  transform: scale(1.05);
}

.info {
  padding: 1rem;
}

.info h3 {
  color: var(--rebel-base);
  font-size: 1.25rem;
  margin-bottom: 1rem;
  text-transform: uppercase;
  letter-spacing: 1px;
}

.info p {
  color: var(--rebel-base);
  margin: 0.5rem 0;
  display: flex;
  justify-content: space-between;
  padding: 0.5rem;
  border-radius: 4px;
  background: rgba(255, 59, 59, 0.1);
  font-size: 0.875rem;
}
