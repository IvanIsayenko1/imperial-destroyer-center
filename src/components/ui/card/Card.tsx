import React from 'react';
import './Card.css';

type CardProps = {
  imageSrc: string;
  name: string;
  properties: { [key: string]: string };
};

export const Card = ({ imageSrc, name, properties }: CardProps) => {
  return (
    <div className="card">
      <div className="image">
        <img src={imageSrc} alt={name} />
      </div>
      <div className="info">
        <h3>{name}</h3>
        {Object.entries(properties).map(([key, value]) => (
          <p key={key}>{`${key}: ${value}`}</p>
        ))}
      </div>
    </div>
  );
};
