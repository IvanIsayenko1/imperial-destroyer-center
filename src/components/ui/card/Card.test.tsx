import { render, screen } from "@testing-library/react";
import { Card } from "./Card";

describe("Card component", () => {
  const testProps = {
    imageSrc: "https://example.com/image.jpg",
    name: "Test Card",
    properties: {
      property1: "value1",
      property2: "value2",
    },
  };

  it("shour render the card correctly", () => {
    render(<Card {...testProps} />);
    expect(screen.getByAltText("Test Card")).toHaveAttribute(
      "src",
      testProps.imageSrc
    );
    expect(screen.getByText("Test Card")).toBeInTheDocument();
    Object.entries(testProps.properties).forEach(([key, value]) => {
      expect(screen.getByText(`${key}: ${value}`)).toBeInTheDocument();
    });
  });

  it("should render the correct number of properties", () => {
    render(<Card {...testProps} />);
    expect(screen.getAllByRole("paragraph")).toHaveLength(
      Object.entries(testProps.properties).length
    );
  });

  it("should render correctly with no properties", () => {
    render(<Card {...{ ...testProps, properties: {} }} />);
    expect(screen.getByText("Test Card")).toBeInTheDocument();
    expect(screen.queryAllByRole("paragraph")).toHaveLength(0);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Card {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
