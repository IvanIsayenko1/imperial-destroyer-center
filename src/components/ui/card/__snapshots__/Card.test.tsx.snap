// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="image"
    >
      <img
        alt="Test Card"
        src="https://example.com/image.jpg"
      />
    </div>
    <div
      class="info"
    >
      <h3>
        Test Card
      </h3>
      <p>
        property1: value1
      </p>
      <p>
        property2: value2
      </p>
    </div>
  </div>
</DocumentFragment>
`;
